pyfakefs/fake_pathlib.py

```diff
@@ -60,10 +60,15 @@ def init_module(filesystem):
     else:
         # in Python 3.12, the flavour is no longer an own class,
         # but points to the os-specific path module (posixpath/ntpath)
-        fake_os = FakeOsModule(filesystem)
-        fake_path = fake_os.path
-        FakePathlibModule.PureWindowsPath._flavour = fake_path
-        FakePathlibModule.PurePosixPath._flavour = fake_path
+        fake_posix_os = FakeOsModule(filesystem)
+        fake_posix_path = fake_posix_os.path
+        FakePathlibModule.PurePosixPath._flavour = fake_posix_path
+        # The Windows path separators must be customized.
+        fake_nt_os = FakeOsModule(filesystem)
+        fake_nt_path = fake_nt_os.path
+        fake_nt_path.sep = "\\"
+        fake_nt_path.altsep = "/"
```

Comment:
Same to be done for Posix paths.

Reply:
E.g.:
```py
        fake_posix_os = FakeOsModule(filesystem)
        fake_posix_path = fake_posix_os.path
        if filesystem.is_windows_fs:
            # The POSIX path separators must be set if running under Windows
            fake_posix_path.sep = "/"
            fake_posix_path.altsep = None
        fake_posix_path = fake_posix_os.path
        FakePathlibModule.PurePosixPath._flavour = fake_posix_path
        fake_nt_os = FakeOsModule(filesystem)
        fake_nt_path = fake_nt_os.path
        if not filesystem.is_windows_fs:
            # The Windows path separators must be set if running under POSIX
            fake_nt_path.sep = "\\"
            fake_nt_path.altsep = "/"
        FakePathlibModule.PureWindowsPath._flavour = fake_nt_path
```
This avoids breaking tests that set the separator, and also handles the reciprocal case.

Reply:
When I added conditionals, the test for #1006 started failing. I'm currently force-pushing changes to try resolving the failing Windows tests.